### PR TITLE
[BXMSPROD-1981] setup nightly

### PR DESCRIPTION
**Pull request by lampajr**
**JIRA**: https://issues.redhat.com/browse/BXMSPROD-1981

**Change Description**:

Setup nightly pipeline for `drools-ansible-rulebook-integration` project.

> NOTE: Also added PR backporting workflow.

**Referenced Pull Requests:**
- https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/267
- https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/15/
- https://github.com/kiegroup/kie-ci/pull/1450

> NOTE: https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/267 must be merged before merging this one

### CODE REVIEW

File: .ci/project-dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: "2.1"
```
Comment (lampajr) [inline .ci/project-dependencies.yaml:1]: I will remove this file as soon as https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/14 is merged

**Review by Ginxo — APPROVED** [review]
LGTM

**Comment by mareknovotny** [comment]
it looks fine

File: .ci/project-dependencies.yaml (hunk 1 above)
Comment (lampajr) [reply .ci/project-dependencies.yaml]: removed `project-dependencies.yaml`

**Review by rgdoliveira — APPROVED** [review]

**Review by bacciotti — APPROVED** [review]

**Pull request merged**